File: CSharp/ILinkai.Weixin.Sample/Examples/AccountExample.cs
using Microsoft.Extensions.Logging;
using ILinkai.Weixin.Sdk.Auth;

namespace ILinkai.Weixin.Sample.Examples;

public class AccountExample : ExampleBase
{
    public override string Name => "account";
    public override string Description => "账户管理示例";
    public override string Usage => "account list";

    public AccountExample(ILoggerFactory loggerFactory) : base(loggerFactory) { }

    public override Task<int> ExecuteAsync(string[] args)
    {
        PrintHeader("账户管理示例");

        var accountStore = new AccountStore();

        if (args.Length < 2 || args[1] != "list")
        {
            Console.WriteLine($"用法: {Usage}");
            return Task.FromResult(1);
        }

        var accounts = accountStore.ListAccountIds();

        if (accounts.Count == 0)
        {
            Console.WriteLine("没有已注册的账户");
            Console.WriteLine("请先运行 'login' 命令进行登录");
            return Task.FromResult(0);
        }

        Console.WriteLine($"已注册账户 ({accounts.Count} 个):");
        Console.WriteLine();

        foreach (var id in accounts)
        {
            var account = accountStore.ResolveAccount(id);
            var accountData = accountStore.LoadAccount(id);

            Console.WriteLine($"账户ID: {account.AccountId}");
            Console.WriteLine($"  基础URL: {account.BaseUrl}");
            Console.WriteLine($"  CDN URL: {account.CdnBaseUrl}");
            Console.WriteLine($"  已配置: {account.Configured}");
            Console.WriteLine($"  已启用: {account.Enabled}");

            if (!string.IsNullOrEmpty(accountData?.Token))
            {
                Console.WriteLine($"  Token: {accountData.Token.Substring(0, Math.Min(20, accountData.Token.Length))}...");
            }

            if (!string.IsNullOrEmpty(accountData?.UserId))
            {
                Console.WriteLine($"  用户ID: {accountData.UserId}");
            }

            if (accountData?.SavedAt != null)
            {
                Console.WriteLine($"  保存时间: {accountData.SavedAt}");
            }

            Console.WriteLine();
        }

        return Task.FromResult(0);
    }
}

File: Python/samples/account_example.py
"""账户管理示例"""
from samples.example_base import ExampleBase
from ilinkai_weixin.auth.account_store import AccountStore


class AccountExample(ExampleBase):
    name = "account"
    description = "账户管理示例"
    usage = "account list"

    def execute(self, args: list) -> int:
        self.print_header("账户管理示例")

        account_store = AccountStore()

        if len(args) < 2 or args[1] != "list":
            print(f"用法: {self.usage}")
            return 1

        accounts = account_store.list_account_ids()

        if not accounts:
            print("没有已注册的账户")
            print("请先运行 'login' 命令进行登录")
            return 0

        print(f"已注册账户 ({len(accounts)} 个):")
        print()

        for id_ in accounts:
            account = account_store.resolve_account(id_)
            account_data = account_store.load_account(id_)

            print(f"账户ID: {account.account_id}")
            print(f"  基础URL: {account.base_url}")
            print(f"  CDN URL: {account.cdn_base_url}")
            print(f"  已配置: {account.configured}")
            print(f"  已启用: {account.enabled}")

            if account_data and account_data.token:
                print(f"  Token: {account_data.token[:20]}...")
            if account_data and account_data.user_id:
                print(f"  用户ID: {account_data.user_id}")
            if account_data and account_data.saved_at:
                print(f"  保存时间: {account_data.saved_at}")
            print()

        return 0
